### PR TITLE
CI: Reuse check schema

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Part of https://github.com/Lightning-AI/lightning/issues/12545.

Same as https://github.com/Lightning-AI/lightning/pull/14469.

## Before submitting
- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes?
- [n/a] Did you write any **new necessary tests**? [not needed for typos/docs]
- [n/a] Did you verify **new and existing tests pass** locally with your changes?
- [n/a] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by krshrimali — APPROVED** [review]
@akihironitta - Just curious, what does this do? Sorry, I don't have much context. Approving as I trust you on this ;)

**Comment by akihironitta** [comment]
> @akihironitta - Just curious, what does this do? Sorry, I don't have much context. Approving as I trust you on this ;)

Haha, thanks for asking 😆

The actual workflow file sits in another repository: https://github.com/Lightning-AI/utilities/blob/215a98333e5fe8906156a290ad5f577e176572d1/.github/workflows/check-schema.yml#L25-L37
, and basically, it just checks if yml config files (such as azure and GHA workflow files) are in the correct structure, e.g. indentation, typo, etc.

For example, here's [the azure schema file](https://raw.githubusercontent.com/microsoft/azure-pipelines-vscode/v1.205.0/service-schema.json) that defines what the correct structure is for azure pipelines' config files.

cc @Borda just in case you have something to add here.

**Comment by krshrimali** [comment]
Thanks, @akihironitta for the context. This helps! So does this mean that we'll have to rename `.azure-pipelines` folder to `.azure_pipelines`? `Check shchema` job is failing in this PR:

```python
File "/home/runner/.local/lib/python3.8/site-packages/identify/identify.py", line [44](https://github.com/Lightning-AI/lightning-flash/runs/8242093713?check_suite_focus=true#step:7:45), in tags_from_path
    raise ValueError(f'{path} does not exist.')
ValueError: .azure_pipelines//*.yml does not exist.
```

We have `.azure-pipelines` folder, so maybe that's not expected?

**Comment by akihironitta** [comment]
Thank you @krshrimali for pointing it out! I have it fixed now :)

**Comment by krshrimali** [comment]
> 

Awesome, thank you so much! 🔥 Let's merge once all relevant tests are green.

**Comment by akihironitta** [comment]
It seems the job added here is failing. Looking into it

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@akihironitta could be that some inputs shall be a string, but they are passed as numbers...

**Comment by akihironitta** [comment]
It was indeed due to the incorrect schema provided by MSFT. `true` and `false` are valid values in the Azure Pipelines service but are considered invalid with the schema file as reported in the linked threads:
- schema definition: https://github.com/microsoft/azure-pipelines-vscode/blob/2a607113c4c0fb844d28ca0582a50920448c0e3e/service-schema.json#L4008
- https://github.com/microsoft/azure-pipelines-vscode/issues/446
- https://developercommunity.visualstudio.com/t/Azure-Pipelines-schema-inconsistently-li/1668070

In short, what we can do now is to quote all `true` and `false` occurrences so that we have `"true"` and `"false"` 😄

**Comment by akihironitta** [comment]
I see another issue around `PublishCodeCoverageResults@1`, and it turned out that this is something @Borda reported to their repos quite a while ago 😄
- https://github.com/microsoft/azure-pipelines-vscode/issues/412
- https://github.com/python-jsonschema/check-jsonschema/issues/10

**Comment by akihironitta** [comment]
@Borda Would you say it's worth removing `PublishCodeCoverageResults@1` to have the schema check pass?

File: .github/workflows/ci-schema.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  push:
+    branches: [master, "release/*"]
+  pull_request:
+    branches: [master, "release/*"]
+
+jobs:
+  check-schema:
+    uses: Lightning-AI/utilities/.github/workflows/check-schema.yml@v0.4.0
+    with:
+      azure-dir: ''
```
Comment (akihironitta) [inline .github/workflows/ci-schema.yml:13]: Given that Microsoft doesn't provide the correct schema file as I commented in https://github.com/Lightning-AI/lightning-flash/pull/1455#issuecomment-1244793607, I'd like to suggest skipping all azure configs for the time being.

**Review by akihironitta — COMMENTED** [review]
@Borda Please have a final look :)

**Review by Borda — APPROVED** [review]
